system_metrics_collector/src/system_metrics_collector/linux_process_cpu_measurement_node.hpp

```diff
@@ -0,0 +1,94 @@
+// Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SYSTEM_METRICS_COLLECTOR__LINUX_PROCESS_CPU_MEASUREMENT_NODE_HPP_
+#define SYSTEM_METRICS_COLLECTOR__LINUX_PROCESS_CPU_MEASUREMENT_NODE_HPP_
+
+#include <chrono>
+#include <cmath>
+#include <string>
+#include <tuple>
+
+#include "periodic_measurement_node.hpp"
+#include "proc_cpu_data.hpp"
+#include "utilities.hpp"
+
+#include "rclcpp/rclcpp.hpp"
+#include "rcutils/logging_macros.h"
```

Comment:
Consider using logging utilities like those in [rosbag2](https://github.com/ros2/rosbag2/blob/master/rosbag2_cpp/include/rosbag2_cpp/logging.hpp)

Reply:
This should be a separate PR.